src/Control/Monad/Bayes/Traced/Common.hs

```diff
@@ -65,21 +65,32 @@ bind dx f = do
   t2 <- f (output t1)
   return $ t2 {variables = variables t1 ++ variables t2, density = density t1 * density t2}
 
--- | A single Metropolis-corrected transition of single-site Trace MCMC.
-mhTrans :: MonadSample m => Weighted (FreeSampler m) a -> Trace a -> m (Trace a)
-mhTrans m t@Trace {variables = us, density = p} = do
+-- | a proposal distribution over randomness
```

Comment:
IMO "a proposal distribution over randomness" is not very descriptive. I don't know what it means.

Reply:
there was some discussion how to call the numbers in the interval 0 1 that are the original source of a probabilistic computation. We settled calling them `randomness :: [Double]` where `Double` really is limited to the interval 0 1 which is not enforced atm. If we call this randomness _r_, a proposal distribution over randomness gives you P(r'|r), the probability to get a _r'_ for a given _r_. In other words it gives you the probability to draw a new proposed sample from a given one in the MCMC chain. In Haskell terms _P(r'|r)_ translates to `[Double] -> m [Double]` where the first argument is _r_ and the second one the conditioned distribution _P(r')_.